networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java

```diff
@@ -74,44 +78,53 @@
-  private int eth2RpcOutstandingPingThreshold = DEFAULT_ETH2_RPC_OUTSTANDING_PING_THRESHOLD;
-  private final Duration eth2StatusUpdateInterval = DEFAULT_ETH2_STATUS_UPDATE_INTERVAL;
-  private Optional<Checkpoint> requiredCheckpoint = Optional.empty();
-  private Spec spec;
-  private OperationProcessor<SignedContributionAndProof>
+  protected P2PConfig config;
+  protected EventChannels eventChannels;
+  protected RecentChainData recentChainData;
+  protected OperationProcessor<SignedBeaconBlock> gossipedBlockProcessor;
+  protected OperationProcessor<ValidateableAttestation> gossipedAttestationConsumer;
+  protected OperationProcessor<ValidateableAttestation> gossipedAggregateProcessor;
+  protected OperationProcessor<AttesterSlashing> gossipedAttesterSlashingConsumer;
+  protected GossipPublisher<AttesterSlashing> attesterSlashingGossipPublisher;
+  protected OperationProcessor<ProposerSlashing> gossipedProposerSlashingConsumer;
+  protected GossipPublisher<ProposerSlashing> proposerSlashingGossipPublisher;
+  protected OperationProcessor<SignedVoluntaryExit> gossipedVoluntaryExitConsumer;
+  protected GossipPublisher<SignedVoluntaryExit> voluntaryExitGossipPublisher;
+  protected ProcessedAttestationSubscriptionProvider processedAttestationSubscriptionProvider;
+  protected StorageQueryChannel historicalChainData;
+  protected MetricsSystem metricsSystem;
+  protected final List<RpcMethod<?, ?, ?>> rpcMethods = new ArrayList<>();
+  protected final List<PeerHandler> peerHandlers = new ArrayList<>();
+  protected TimeProvider timeProvider;
+  protected AsyncRunner asyncRunner;
+  protected KeyValueStore<String, Bytes> keyValueStore;
+  protected Duration eth2RpcPingInterval = DEFAULT_ETH2_RPC_PING_INTERVAL;
+  protected int eth2RpcOutstandingPingThreshold = DEFAULT_ETH2_RPC_OUTSTANDING_PING_THRESHOLD;
+  protected final Duration eth2StatusUpdateInterval = DEFAULT_ETH2_STATUS_UPDATE_INTERVAL;
+  protected Optional<Checkpoint> requiredCheckpoint = Optional.empty();
+  protected Spec spec;
+  protected OperationProcessor<SignedContributionAndProof>
       gossipedSignedContributionAndProofProcessor;
-  private GossipPublisher<SignedContributionAndProof> signedContributionAndProofGossipPublisher;
-  private OperationProcessor<ValidateableSyncCommitteeMessage>
+  protected GossipPublisher<SignedContributionAndProof> signedContributionAndProofGossipPublisher;
+  protected OperationProcessor<ValidateableSyncCommitteeMessage>
       gossipedSyncCommitteeMessageProcessor;
-  private GossipPublisher<ValidateableSyncCommitteeMessage> syncCommitteeMessageGossipPublisher;
+  protected GossipPublisher<ValidateableSyncCommitteeMessage> syncCommitteeMessageGossipPublisher;
+
+  protected Supplier<DiscoveryNetworkBuilder> discoveryNetworkBuilderSupplier =
+      DiscoveryNetworkBuilder::create;
+  protected Supplier<LibP2PNetworkBuilder> libP2PNetworkBuilderSupplier =
+      LibP2PNetworkBuilder::create;
```

Comment:
since we're using protected methods so expecting a class to override this anyway, would it make more sense to just make these abstract methods?

Reply:
Agree. Was actually bouncing between these two approaches.